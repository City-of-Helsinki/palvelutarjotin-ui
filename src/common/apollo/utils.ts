import { NormalizedCacheObject, ApolloClient } from '@apollo/client';

export class MutableReference<Ref = unknown> {
  _value?: Ref;

  constructor(initialReference?: Ref) {
    this._value = initialReference;
  }

  get reference(): Ref | undefined {
    return this._value;
  }

  set reference(ref: Ref | undefined) {
    this._value = ref;
  }
}

type InitApolloClientConfig<
  TCacheShape,
  Client extends ApolloClient<TCacheShape>,
> = {
  initialState: TCacheShape;
  mutableCachedClient: MutableReference<Client>;
  createClient: () => Client;
};

// https://www.apollographql.com/blog/building-a-next-js-app-with-apollo-client-slash-graphql/
export function initializeApolloClient<
  TCacheShape = NormalizedCacheObject,
  Client extends ApolloClient<TCacheShape> = ApolloClient<TCacheShape>,
>({
  initialState,
  mutableCachedClient,
  createClient,
}: InitApolloClientConfig<TCacheShape, Client>): ApolloClient<TCacheShape> {
  const _apolloClient = mutableCachedClient.reference ?? createClient();

  // Initial state hydration
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();

    // Restore the cache using the data passed from
    // getStaticProps/getServerSideProps combined with the existing cached data
    _apolloClient.cache.restore({ ...existingCache, ...initialState });
  }

  // For SSG and SSR always create a new Apollo Client
  if (typeof window === 'undefined') {
    return _apolloClient;
  }

  // Create the Apollo Client once in the client
  if (!mutableCachedClient.reference) {
    mutableCachedClient.reference = _apolloClient;
  }

  return _apolloClient;
}
