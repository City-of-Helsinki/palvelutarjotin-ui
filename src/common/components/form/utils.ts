/* eslint-disable @typescript-eslint/no-explicit-any */
import { FormikErrors, FormikTouched } from 'formik';
import { TFunction } from 'i18next';

/** Get error text
 * @param {Object} errors
 * @param {Object} touched
 * @param {string} name
 * @param {Function} t
 * @return {string}
 */
export const getErrorText = (
  errors: FormikErrors<any>,
  touched: FormikTouched<any>,
  name: string,
  t: TFunction
): string => {
  const error: any = errors[name];

  return !!error && touched[name]
    ? typeof error === 'string'
      ? t(error)
      : t(error.key, error)
    : '';
};
