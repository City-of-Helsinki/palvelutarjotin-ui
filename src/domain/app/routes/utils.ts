import { CollectionItemType } from 'react-helsinki-headless-cms';

import { ROUTES } from './constants';

export const getCmsPagePath = (uri?: string | null): string => {
  return ROUTES.CMS_PAGE.replace('/:uri', uri ?? '');
};

export const getCmsArticlePath = (uri?: string | null): string => {
  return ROUTES.CMS_ARTICLE.replace('/:uri', uri ?? '');
};

export const getCollectionItemUrl = (item: CollectionItemType): string => {
  if (!item) {
    return '#';
  }
  if (item.__typename === 'Post') {
    return getCmsArticlePath(item.uri);
  }
  if (item.__typename === 'Page') {
    return getCmsPagePath(item.uri);
  }
  return item?.uri ?? '';
};
