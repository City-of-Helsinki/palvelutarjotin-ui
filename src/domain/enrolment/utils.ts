import {
  EnrolOccurrenceMutationInput,
  Language,
  NotificationType,
} from '../../generated/graphql';
import { EnrolmentFormFields } from './enrolmentForm/constants';

const getNotificationType = (values: EnrolmentFormFields): NotificationType => {
  return values.hasEmailNotification && values.hasSmsNotification
    ? NotificationType.EmailSms
    : NotificationType.Email;
};

export const getEnrolmentPayload = ({
  occurrenceIds,
  values,
}: {
  occurrenceIds: string[];
  values: EnrolmentFormFields;
}): EnrolOccurrenceMutationInput => {
  return {
    occurrenceIds,
    notificationType: getNotificationType(values),
    person: values.isSameResponsiblePerson
      ? undefined
      : { ...values.person, language: values.language as Language },
    studyGroup: {
      ...values.studyGroup,
      amountOfAdult: Number(values.studyGroup.amountOfAdult),
      groupSize: Number(values.studyGroup.groupSize),
      person: {
        ...values.studyGroup.person,
        language: values.language as Language,
      },
      studyLevels: values.studyGroup.studyLevels,
    },
  };
};

declare let grecaptcha: ReCAPTCHA;

interface ReCAPTCHA {
  ready: (f: () => void) => void;
  execute: (key: string, action: { action: string }) => Promise<string>;
}

export const getCAPTCHAToken = (): Promise<string> => {
  return new Promise((resolve, reject) => {
    if (process.env.NEXT_PUBLIC_CAPTCHA_KEY) {
      const captchaKey = process.env.NEXT_PUBLIC_CAPTCHA_KEY;
      grecaptcha.ready(() => {
        grecaptcha
          .execute(captchaKey, { action: 'submit' })
          .then(resolve)
          .catch(reject);
      });
    } else {
      reject('CAPTCHA_KEY missing!');
    }
  });
};
