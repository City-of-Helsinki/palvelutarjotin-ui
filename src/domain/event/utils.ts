import { isToday, isTomorrow } from 'date-fns';
import { TFunction } from 'next-i18next';

import { EventFieldsFragment } from '../../generated/graphql';
import { Language } from '../../types';
import formatDate from '../../utils/formatDate';
import getLocalisedString from '../../utils/getLocalisedString';
import getTimeFormat from '../../utils/getTimeFormat';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';

/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

export const getEventStartTimeStr = (
  event: EventFieldsFragment,
  locale: Language,
  t: TFunction
): string | null => {
  const startTime = event.startTime ? new Date(event.startTime) : null;
  const timeFormat = getTimeFormat(locale);

  if (!startTime) return null;

  if (isToday(startTime))
    return t('event:eventCard.startTime.today', {
      time: formatDate(startTime, timeFormat, locale),
    });

  if (isTomorrow(startTime))
    return t('event:eventCard.startTime.tomorrow', {
      time: formatDate(startTime, timeFormat, locale),
    });

  return t('event:eventCard.startTime.other', {
    date: formatDate(startTime),
    time: formatDate(startTime, timeFormat, locale),
  });
};

/**
 * Check is event free
 * @param eventData
 * @return {boolean}
 */
export const isEventFree = (event: EventFieldsFragment): boolean =>
  Boolean(event.offers.find((item) => item.isFree)?.isFree);

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const getEventFields = (
  event: EventFieldsFragment | undefined | null,
  locale: Language
) =>
  event
    ? {
        isEventFree: isEventFree(event),
        eventName: getLocalisedString(event?.name, locale),
        shortDescription: getLocalisedString(event?.shortDescription, locale),
        description: getLocalisedString(event?.description, locale),
        imageUrl: event?.images?.[0]?.url,
        imageAltText: event?.images?.[0]?.altText,
        photographerName: event?.images?.[0]?.photographerName,
        organization: event.pEvent?.organisation?.name,
        contactPhoneNumber: event.pEvent?.contactPhoneNumber,
        contactEmail: event.pEvent?.contactEmail,
        contactPerson: event.pEvent?.contactPerson?.name,
      }
    : {};
