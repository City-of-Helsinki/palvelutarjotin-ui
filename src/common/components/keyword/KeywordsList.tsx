import { UrlObject } from 'url';

import * as React from 'react';
import { useTranslation } from 'react-i18next';

import { ROUTES } from '../../../domain/app/routes/constants';
import { KEYWORD_QUERY_PARAMS } from '../../../domain/events/constants';
import {
  KeywordOptions,
  useKeywordOptions,
} from '../../../domain/events/eventSearchForm/useKeywordOptions';
import { KeywordFieldsFragment } from '../../../generated/graphql';
import useLocale from '../../../hooks/useLocale';
import { Language } from '../../../types';
import getLocalisedString from '../../../utils/getLocalisedString';
import Keyword from './Keyword';
import styles from './keyword.module.scss';

type KeywordsListProps = {
  keywords: KeywordFieldsFragment[];
};

const KeywordsList: React.FC<KeywordsListProps> = ({ keywords }) => {
  const locale = useLocale();
  const { t } = useTranslation();
  const keywordOptions = useKeywordOptions();
  const keywordsPropArr = getKeywordsProps(keywords, keywordOptions, locale);

  return (
    <ul className={styles.keywordsList}>
      {keywordsPropArr.map((k) => (
        <li key={k.id}>
          <Keyword
            href={{ pathname: ROUTES.HOME, query: k.query }}
            keyword={k.label}
            aria-label={t('event:keywords.labelKeywordLink', {
              keyword: k.label,
            })}
          />
        </li>
      ))}
    </ul>
  );
};

// Kultus uses many keyword sets, so keywords need to be mapped to correct keyword query parameter
// E.g. Kirjastovierailu -> "?additionalCriteria=kultus%3A20", Jooga -> "?categories=yso%3Ap3111"
const getKeywordsProps = (
  keywords: KeywordFieldsFragment[],
  keywordOptions: KeywordOptions,
  locale: Language
): { label: string; query: UrlObject['query']; id: string }[] => {
  const keywordTypeMappers: {
    keywordOptionsKey: keyof KeywordOptions;
    queryKey: string;
  }[] = [
    {
      queryKey: KEYWORD_QUERY_PARAMS.ADDITIONAL_CRITERIA,
      keywordOptionsKey: 'additionalCriteriaKeywords',
    },
    {
      queryKey: KEYWORD_QUERY_PARAMS.TARGET_GROUPS,
      keywordOptionsKey: 'targetGroups',
    },
    {
      queryKey: KEYWORD_QUERY_PARAMS.CATEGORIES,
      keywordOptionsKey: 'categoryKeywords',
    },
  ];

  // match keywords from keyword sets to get correct query parameter
  return keywords.map((keyword) => {
    const label = getLocalisedString(keyword.name || {}, locale);

    let keywordProps:
      | { label: string; query: UrlObject['query']; id: string }
      | undefined;

    keywordTypeMappers.every(({ keywordOptionsKey, queryKey }) => {
      if (
        keywordOptions[keywordOptionsKey].find((k) => k.value === keyword.id)
      ) {
        keywordProps = {
          id: keyword.id ?? '',
          query: {
            [queryKey]: keyword.id ? [keyword.id] : null,
          },
          label,
        };
        return false;
      }
      return true;
    });

    return keywordProps
      ? keywordProps
      : {
          id: keyword.id ?? '',
          query: {
            categories: keyword.id ? [keyword.id] : null,
          },
          label,
        };
  });
};

export default KeywordsList;
