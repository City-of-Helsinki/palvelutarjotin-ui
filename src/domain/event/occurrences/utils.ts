import {
  EventFieldsFragment,
  OccurrenceFieldsFragment,
} from '../../../generated/graphql';
import { ENROLMENT_ERRORS } from '../../enrolment/constants';
import {
  hasOccurrenceSpace,
  isEnrolmentClosed,
  isOccurrenceCancelled,
} from '../../occurrence/utils';

export function getEnrolmentError(
  occurrence: OccurrenceFieldsFragment,
  event: EventFieldsFragment
): ENROLMENT_ERRORS | null {
  if (isEnrolmentClosed(occurrence, event))
    return ENROLMENT_ERRORS.ENROLMENT_CLOSED_ERROR;
  if (!hasOccurrenceSpace(occurrence))
    return ENROLMENT_ERRORS.NOT_ENOUGH_CAPACITY_ERROR;
  if (isOccurrenceCancelled(occurrence)) {
    return ENROLMENT_ERRORS.ENROLMENT_CANCDELLED_ERROR;
  }

  return null;
}

export function getOrderedLanguages(
  languages: { id: string; name: string }[]
): string[] {
  const languageOrder = ['fi', 'sv', 'en'];
  const languageStrings = languages.map((l) => l.id.split('_')[0]);

  // order languages based on languageOrder, other languages go
  // to the end of the array
  return languageStrings.sort((a, b) => {
    const aIndex = languageOrder.indexOf(a);
    const bIndex = languageOrder.indexOf(b);

    if (aIndex > -1 && bIndex > -1) {
      return aIndex - bIndex;
    }

    if (aIndex > -1 && bIndex === -1) {
      return -1;
    }

    if (bIndex > -1 && aIndex === -1) {
      return 1;
    }

    return 0;
  });
}
