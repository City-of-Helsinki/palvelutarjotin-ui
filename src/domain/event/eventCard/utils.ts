import { TFunction } from 'next-i18next';

import { EnrolmentStatusKeywordPropsType } from '../../../common/components/keyword/EnrolmentStatusKeyword';
import { EventsFieldsFragment } from '../../../generated/graphql';
import { Language } from '../../../types';
import { formatDateRange } from '../../../utils/time/format';
import { isMultidayOccurrence } from '../../occurrence/utils';
import { getEventStartTimeStr } from '../utils';

export const getOccurrenceDateStr = (
  occurrence: { startTime: string; endTime: string },
  locale: Language,
  t: TFunction
): string | null => {
  if (isMultidayOccurrence(occurrence)) {
    return formatDateRange(
      new Date(occurrence.startTime),
      new Date(occurrence.endTime)
    );
  }

  return getEventStartTimeStr(new Date(occurrence.startTime), locale, t);
};

/**
 * Has the event enrolments enabled in any of it's occurrences?
 * @param event The event to check the status for.
 * @returns `true` if any of the event occurrences has enrolments and
 *  `false` if none.
 */
export const hasEventEnrolmentsEnabled = (event: EventsFieldsFragment) =>
  Boolean(event.pEvent.enrolmentStart) || hasEventExternalEnrolments(event);

/**
 * Has the event external enrolments enabled in any of it's occurrences?
 *
 * @param event The event to check the status for.
 * @returns `true` if the event has some external enrolments and
 *  `false` if none of the occurrences has external enrolments.
 */
export const hasEventExternalEnrolments = (event: EventsFieldsFragment) =>
  Boolean(event.pEvent.externalEnrolmentUrl);

/**
 * Resolve the enrolment status of an event. Has the event free spots available?
 *
 * @param event The event to check the status for.
 * @returns Returns `null` if the enrolment status could not be resolved or
 *    the event has no enrolments at all,
 *    `true` if there are any free spots available and `false` if there are no
 *    spots available.
 */
export const hasEventFreeSpotsAvailable = (
  event: EventsFieldsFragment
): boolean | null => event.pEvent.hasSpaceForEnrolments ?? null;

/**
 * Determines the enrollment status of an event.
 *
 * @param event The event to check the enrollment status for.
 * @returns The enrollment status of the event, or `undefined`
 *      if no enrolments exist or the status could not be determined.
 */
export const getEventEnrolmentStatus = (
  event: EventsFieldsFragment
): EnrolmentStatusKeywordPropsType['enrolmentStatus'] => {
  if (hasEventEnrolmentsEnabled(event)) {
    if (hasEventExternalEnrolments(event)) {
      return 'externally_unknown';
    }
    const hasEventFreeSpots = hasEventFreeSpotsAvailable(event);

    if (hasEventFreeSpots === false) {
      // Event has no free spots available, so it is full.
      return 'full';
    } else if (hasEventFreeSpots === true) {
      // Event has some free spots available.
      return 'free_space_available';
    }
  }
  // No enrolments at all or the enrolment status could not be resolved
  return undefined;
};
