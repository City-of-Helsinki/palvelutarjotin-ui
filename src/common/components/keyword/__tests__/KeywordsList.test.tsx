import { MockedResponse } from '@apollo/client/testing';
import * as React from 'react';

import {
  Keyword,
  KeywordSetDocument,
  KeywordSetType,
  LocalisedObject,
} from '../../../../generated/graphql';
import {
  fakeKeyword,
  fakeLocalizedObject,
} from '../../../../utils/mockDataUtils';
import { render, act, screen } from '../../../../utils/testUtils';
import KeywordsList from '../KeywordsList';

const categoryKeywords = [
  { id: 'kultus:1', name: 'Vaikuttaminen ja osallisuus' },
  { id: 'kultus:2', name: 'Teatteri, tanssi ja sirkus' },
  { id: 'kultus:3', name: 'Musiikki' },
  { id: 'kultus:4', name: 'Sanataide ja kirjallisuus' },
  { id: 'kultus:5', name: 'Tiede' },
];

const additionCriteriaKeywords = [
  { id: 'kultus:6', name: 'Työpaja' },
  { id: 'kultus:7', name: 'Luento' },
  { id: 'kultus:8', name: 'Luontokoulu' },
  { id: 'kultus:9', name: 'Kirjastovierailu' },
  { id: 'kultus:10', name: 'Konsertti' },
];

const targetGroupKeywords = [
  { id: 'kultus:11', name: 'Valmistava opetus' },
  { id: 'kultus:12', name: '1.–2. luokat' },
  { id: 'kultus:13', name: '3.–6. luokat' },
  { id: 'kultus:14', name: '7.–9. luokat' },
  { id: 'kultus:15', name: 'lukio ja ammatillinen opetus' },
];

const getKeywordSetResult = ({
  keywords,
  name,
  internalId,
}: {
  keywords: Keyword[];
  name: LocalisedObject;
  internalId: string;
}) => {
  return {
    data: {
      keywordSet: {
        keywords,
        name,
        internalId,
      },
    },
  };
};

const mocks: MockedResponse[] = [
  {
    request: {
      query: KeywordSetDocument,
      variables: { setType: KeywordSetType.AdditionalCriteria },
    },
    result: getKeywordSetResult({
      internalId:
        'https://api.hel.fi/linkedevents-test/v1/keyword_set/kultus:additional_criteria/',
      name: fakeLocalizedObject('Kultus-kategoriat'),
      keywords: additionCriteriaKeywords.map((k) =>
        fakeKeyword({ id: k.id, name: fakeLocalizedObject(k.name) })
      ),
    }),
  },
  {
    request: {
      query: KeywordSetDocument,
      variables: { setType: KeywordSetType.Category },
    },
    result: getKeywordSetResult({
      internalId:
        'https://api.hel.fi/linkedevents-test/v1/keyword_set/kultus:categories/',
      name: fakeLocalizedObject('Kultus-kategoriat'),
      keywords: categoryKeywords.map((k) =>
        fakeKeyword({ id: k.id, name: fakeLocalizedObject(k.name) })
      ),
    }),
  },
  {
    request: {
      query: KeywordSetDocument,
      variables: { setType: KeywordSetType.TargetGroup },
    },
    result: getKeywordSetResult({
      internalId:
        'https://api.hel.fi/linkedevents-test/v1/keyword_set/kultus:target_groups/',
      name: fakeLocalizedObject('Kultus Kohderyhmä'),
      keywords: targetGroupKeywords.map((k) =>
        fakeKeyword({ id: k.id, name: fakeLocalizedObject(k.name) })
      ),
    }),
  },
];

it('renders list of keywords as static texts', async () => {
  const { container } = render(
    <KeywordsList
      identifier="test-identifier"
      keywords={[
        ...categoryKeywords,
        ...targetGroupKeywords,
        ...additionCriteriaKeywords,
      ].map((k) =>
        fakeKeyword({ id: k.id, name: fakeLocalizedObject(k.name) })
      )}
    />,
    { mocks }
  );

  await act(() => new Promise((res) => setTimeout(res, 1000)));

  const categoryKeyword = screen.getByText(/vaikuttaminen ja osallisuus/i);
  expect(categoryKeyword).toBeInTheDocument();

  const targetGroupKeyword = screen.getByText(/valmistava opetus/i);
  expect(targetGroupKeyword).toBeInTheDocument();

  const additionalCriteriaKeyword = screen.getByText(/työpaja/i);
  expect(additionalCriteriaKeyword).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
