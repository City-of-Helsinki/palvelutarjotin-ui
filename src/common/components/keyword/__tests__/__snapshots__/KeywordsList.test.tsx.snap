// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of keywords as links with correct hrefs 1`] = `
<div>
  <ul
    class="keywordsList"
  >
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Vaikuttaminen ja osallisuus"
        class="keyword"
        href="/search?categories=kultus%3A1"
      >
        Vaikuttaminen ja osallisuus
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Teatteri, tanssi ja sirkus"
        class="keyword"
        href="/search?categories=kultus%3A2"
      >
        Teatteri, tanssi ja sirkus
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Musiikki"
        class="keyword"
        href="/search?categories=kultus%3A3"
      >
        Musiikki
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Sanataide ja kirjallisuus"
        class="keyword"
        href="/search?categories=kultus%3A4"
      >
        Sanataide ja kirjallisuus
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Tiede"
        class="keyword"
        href="/search?categories=kultus%3A5"
      >
        Tiede
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Valmistava opetus"
        class="keyword"
        href="/search?targetGroups=kultus%3A11"
      >
        Valmistava opetus
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla 1.–2. luokat"
        class="keyword"
        href="/search?targetGroups=kultus%3A12"
      >
        1.–2. luokat
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla 3.–6. luokat"
        class="keyword"
        href="/search?targetGroups=kultus%3A13"
      >
        3.–6. luokat
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla 7.–9. luokat"
        class="keyword"
        href="/search?targetGroups=kultus%3A14"
      >
        7.–9. luokat
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla lukio ja ammatillinen opetus"
        class="keyword"
        href="/search?targetGroups=kultus%3A15"
      >
        Lukio ja ammatillinen opetus
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Työpaja"
        class="keyword"
        href="/search?additionalCriteria=kultus%3A6"
      >
        Työpaja
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Luento"
        class="keyword"
        href="/search?additionalCriteria=kultus%3A7"
      >
        Luento
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Luontokoulu"
        class="keyword"
        href="/search?additionalCriteria=kultus%3A8"
      >
        Luontokoulu
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Kirjastovierailu"
        class="keyword"
        href="/search?additionalCriteria=kultus%3A9"
      >
        Kirjastovierailu
      </a>
    </li>
    <li>
      <a
        aria-label="Hae tapahtumia avainsanalla Konsertti"
        class="keyword"
        href="/search?additionalCriteria=kultus%3A10"
      >
        Konsertti
      </a>
    </li>
  </ul>
</div>
`;
